### ISSUE TITLE
BUG: Dipole.data attribute not created when using `read_dipole()` for certain files

### ISSUE THREAD

**Issue by ntolley** [jonescompneurolab/hnn-core#520]
Found when trying to reproduce HNN-GUI tutorials. Reproduced via:
```py
from urllib.request import urlretrieve
from hnn_core import read_dipole
data_url = ('https://raw.githubusercontent.com/jonescompneurolab/hnn/master/data/MEG_detection_data/S1_ongoing.txt')
urlretrieve(data_url, 'S1_ongoing.txt')
exp_dpl = read_dipole('S1_ongoing.txt')
exp_dpl.plot()
```
or...
```py
data_url = ('https://raw.githubusercontent.com/jonescompneurolab/hnn/master/data/MEG_detection_data/no_trial_S1_ERP_all_avg.txt')
urlretrieve(data_url, 'no_trial_S1_ERP_all_avg.txt')
exp_dpl = read_dipole('no_trial_S1_ERP_all_avg.txt')
exp_dpl.plot()
```
![image](https://user-images.githubusercontent.com/55253912/180619105-f2b8ba1e-8839-4fb2-b51a-958bd5e1dd23.png)
 
Indeed for some reason the attribute is not created for this object.

**Comment by rythorpe** [comment]
I'm guessing this has to do with the fact that this dipole file only has 2 columns

**Comment by aritrasinha108** [comment]
@jasmainak @rythorpe can I work on this?

**Comment by aritrasinha108** [comment]
> You're sure the data array used to instantiate the Dipole object has shape `(600, 1000)`? That's a huge array and somewhat surprising....

@rythorpe I tried to print the shape `dpl_data` that is being read inside the `read_dipole()` method from the file. 
![Screenshot from 2023-03-08 03-36-54](https://user-images.githubusercontent.com/55765656/223564646-e23a4d43-25da-4317-b710-9a4af0d4a32a.png)

And then ran the first script mentioned above in the description
![Screenshot from 2023-03-08 03-36-08](https://user-images.githubusercontent.com/55765656/223564663-d7ca4c50-a2cb-4474-8f52-742769291513.png)

Got this as the output. The shape is given to be `(600, 1001)`
![Screenshot from 2023-03-08 03-36-35](https://user-images.githubusercontent.com/55765656/223564659-ccf019bb-d805-493d-ab6e-3a3da39edbb0.png)


> Also, @aritrasinha108 let's get your other PR completed first before starting another.

Yeah, sure, I'll finish up my previous PR and then only look into it.

**Comment by rythorpe** [comment]
Sure enough. I don't think I've worked with the `S1_ongoing.txt` data file before which turns out to be quite larger than the others. The problem is that the data getting passed to the `Dipole` constructor doesn't meet any of expected [shape requirements](https://github.com/jonescompneurolab/hnn-core/blob/b9a09d8e8f05e6462f76c119814baa716ddd23d1/hnn_core/dipole.py#L276). We should add appropriate checks that throw an error when the expected read-in data shape is violated and discuss reformating the `S1_ongoing.txt` data file.

**Comment by aritrasinha108** [comment]
@rythorpe can the checks be added as a fix for this issue? please let me know if  its something I can work on

**Comment by rythorpe** [comment]
First we'll need to decide how to handle the current `S1_ongoing.txt` file, since it's formatting is an outlier among the other experimental dipole .txt files and represents a rare use-case that we probably don't want to support. I'd wait on this until @jasmainak @ntolley @chenghuzi can weigh in.

**Comment by jasmainak** [comment]
I would suggest starting with the other file `no_trial_S1_ERP_all_avg.txt` in the PR description since that is the usual format I have seen. Which tutorial has the `S1_ongoing.txt` file @ntolley ?

**Comment by jasmainak** [comment]
close issue?

**Comment by rythorpe** [comment]
Closing in favor of #617
